lib/inc/pxp-agent/modules/download_file.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef SRC_MODULES_DOWNLOAD_FILE_H_
+#define SRC_MODULES_DOWNLOAD_FILE_H_
+
+#include <pxp-agent/module.hpp>
+#include <pxp-agent/module_cache_dir.hpp>
+#include <pxp-agent/action_response.hpp>
+#include <pxp-agent/results_storage.hpp>
+#include <pxp-agent/util/purgeable.hpp>
+#include <pxp-agent/util/bolt_module.hpp>
+
+ #include <cpp-pcp-client/util/thread.hpp>
+
+#include <leatherman/curl/client.hpp>
+#include <set>
+
+namespace PXPAgent {
+namespace Modules {
+
+class DownloadFile : public PXPAgent::Util::BoltModule {
+  public:
+    DownloadFile(const std::vector<std::string>& master_uris,
+                 const std::string& ca,
+                 const std::string& crt,
+                 const std::string& key,
+                 const std::string& proxy,
+                 uint32_t download_connect_timeout,
+                 uint32_t download_timeout,
+                 std::shared_ptr<ModuleCacheDir> module_cache_dir,
+                 std::shared_ptr<ResultsStorage> storage);
+
+   private:
+    boost::filesystem::path exec_prefix_;
+
+     std::vector<std::string> master_uris_;
+
+     uint32_t file_download_connect_timeout_, file_download_timeout_;
+
+     leatherman::curl::client client_;
+    ActionResponse callAction(const ActionRequest& request) override;
+
+     // Since DownloadFile overrides callAction here's no reason to define
+    // buildCommandObject (since it will never be called)
```

Comment:
Maybe this doesn't need to be pure virtual in the base class then -- what if we change things so that `buildCommandObject` returns a pointer to a `CommandObject` instead of an instance, and the base class implementation could just return a `nullptr` (and not try to execute anything in that case)?

Reply:
Yeah, that might make more sense

Reply:
Actually, I think I might rather have it this way.

Download file is the _only_ one of the various actions that won't define buildCommandObject, so I think that might make it okay for this since download is the exception instead of the rule

Reply:
I also think this empty function serves as a sort of historical reference that download file is different from the other actions, since it doesn't perform any command execution.